nilearn/connectome/group_sparse_cov.py

```diff
@@ -957,13 +966,21 @@ def fit(self, subjects, y=None):
             alpha_1, _ = compute_alpha_max(emp_covs, n_samples)
             alpha_0 = 1e-2 * alpha_1
             alphas = np.logspace(np.log10(alpha_0), np.log10(alpha_1),
-                               n_alphas)[::-1]
+                                 n_alphas)[::-1]
 
         covs_init = itertools.repeat(None)
+
+        # Copying the cv generators to use them n_refinements times.
+        cv_list = []
+        cv_ = izip(*cv)
+        for i in range(n_refinements):
+            _, cv_ = itertools.tee(cv_)
+            cv_list.append(cv_)
```

Comment:
Wouldn't it be a better strategy to do this copy in the loop below? Also using the tee maybe?